# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [1]:
a = 101
b = 210
c = 111
d = 402

decision = ((a + b) * c) / d
decision

85.8731343283582

Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [2]:
s1 = 11283
s2 = 89
ostacha = s1 % s2
ostacha

69

In [4]:
#або
11283%89

69

3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [5]:
11283//89

126

In [6]:
#або
c1 = 11283
c2 = 89
tsilykh = c1 // c2
tsilykh

126

4. Запишіть у змінну `name` ваше імʼя.

In [7]:
name = "Карина"
name

'Карина'

5. Запишіть у змінну `surname` ваше прізвище.

In [8]:
surname = "Аветісова"
surname

'Аветісова'

6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [11]:
name_surname = name + " " + surname
name_surname

'Карина Аветісова'

Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [12]:
print(name_surname.upper())

КАРИНА АВЕТІСОВА


Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [18]:
name_surname = "Карина Аветісова"
name_surname [2]

'р'

Очікувана відповідь (тільки з вашими даними): 'n'

9. Виведіть першу та останню літеру вашого прізвища.

In [23]:
name_surname [7], name_surname [15]

('А', 'а')

In [25]:
#або
surname [0], surname [8]

('А', 'а')

Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [26]:
len(surname)

9

Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [28]:
mad = 'Antananarivo'
count_a = print(mad.lower().count("a"))
count_a

4


Очікувана відповідь: 4

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [29]:
my_string = 'The quick brown fox jumps over the lazy dog!!!'
print(my_string.startswith("The"))
print(my_string.endswith("dog!"))

True
False


13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [31]:
print(my_string.split(" "))
print("_".join(['The', 'quick', 'brown', 'fox', 'jumps', 'over', 'the', 'lazy', 'dog!!!']))

['The', 'quick', 'brown', 'fox', 'jumps', 'over', 'the', 'lazy', 'dog!!!']
The_quick_brown_fox_jumps_over_the_lazy_dog!!!


Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`